install-and-configure/install/multi-cluster/federated-etl/aggregator.md

```diff
@@ -0,0 +1,139 @@
+Aggregator serves a critical subset of Kubecost APIs, but will eventually be the default model for Kubecost and serve all APIs. Currently, Aggregator all major monitoring and savings APIs, and also budgets and reporting.
+
+{% hint style="info" %}
+Existing documentation for Kubecost APIs will use endpoints for non-Aggregator environments unless otherwise specified, but will still be compatible after configuring Aggregator.
+{% endhint %}
+
+Aggregator is designed to accommodate queries of large-scale datasets, including reduced load times and potential errors in the UI. It is not designed to introduce new functionality; it is meant to improve functionality at scale.
+
+Aggregator is free for all users to configure, and is always able to be rolled back.
+
+## Configuring Aggregator
+
+### Prerequisites
+
+* Aggregator can only be configured in a Federated ETL environment
+* Must be using v1.107.0 of Kubecost or newer
+* Your *values.yaml* file must have set `kubecostDeployment.queryServiceReplicas` to its default value `0`. 
+
+### Tutorial
+
+Select from one of the two templates below and save the content as *aggregator.yaml*. This will be your configuration template required to set up Aggregator.
+
+Basic configuration:
+
+```
+kubecostAggregator:
+  replicas: 1
+  enabled: true
+  cloudCost:
+    enabled: true
+federatedETL:
+  federatedCluster: true
+kubecostModel:
+  containerStatsEnabled: true
+  federatedStorageConfigSecret: federated-store
+kubecostProductConfigs:
+  clusterName: YOUR_CLUSTER_NAME
+  cloudIntegrationSecret: cloud-integration
+  productKey:
+    enabled: true
+    key: YOUR_KEY
+prometheus:
+  server:
+    global:
+      external_labels:
+        cluster_id: YOUR_CLUSTER_NAME
+# when using managed identity/irsa, set the service account accordingly:
+serviceAccount:
+  create: false
+  name: kubecost-irsa-sa
+```
+
+Advanced configuration (for larger deployments):
+
+```
+kubecostAggregator:
+  replicas: 1
+  enabled: true
+  cloudCost:
+    enabled: true
+  env:
+    # governs parallelism of derivation step
+    # more threads speeds derivation, but requires significantly more 
+    # disk space for WAL storage
+    # default: 6
+    NUM_DERIVER_THREADS: 6
+    # log level
+    # default: info
+    LOG_LEVEL: debug
+  aggregatorDbStorage:
+    # governs storage size of aggregator DB storage
+    # !!NOTE!! disk performance is _critically important_ to aggregator performance
+    # ensure disk is specd high enough, and check for bottlenecks
+    # default: 128Gi
+    storageRequest: 128Gi
+  # tracing, starts embedded jaeger pod
+  # port forward to 16686 on the pod to access jaeger UI 
+  # provides instrumented traces used for troubleshooting performance issues
+  jaeger:
+    enabled: true
+    image: jaegertracing/all-in-one
+    imageVersion: latest
+federatedETL:
+  federatedCluster: true
+kubecostModel:
+  containerStatsEnabled: true

```

Comment:
Suggest adding the following helm values to disable CloudCost in the cost-analyzer pod.

```suggestion
  containerStatsEnabled: true
  cloudCost:
    enabled: false
```

Reply:
@ameijer can you confirm if .Values.kubecostModel.containerStatsEnabled is used?

Should we consider making this the default?

Reply:
I believe container stats is still used in the helm chart and code

Reply:
`.Values.kubecostModel.containerStatsEnabled` is a thing but remains optional. I like it as a default-on because it enables quantile request sizing queries and AFAIK it has been fairly stable in customer envs. It won't matter too much for now (because of https://kubecost.atlassian.net/jira/polaris/projects/KC/ideas/view/2936873?selectedIssue=KC-170) but it would be nice to have the historical data in S3 whenever that work gets completed.